components/sup/src/manager/service_updater.rs

```diff
@@ -88,8 +89,9 @@ impl ServiceUpdater {
                 self.states
                     .entry(service.service_group.clone())
                     .or_insert_with(|| {
-                        let rx = Worker::new(service).start(&service.service_group, None);
-                        UpdaterState::AtOnce(rx)
+                        let (kill_tx, kill_rx) = channel();
```

Comment:
Not for this PR, but it seems like there's an abstraction hiding in here that we might be able to extract... some kind of "kill switch". There's two instances of this in this PR alone, I've got a new one I added in my current work, and there will likely be instances in the future where this pattern could come in handy.

Worth pondering, at any rate 🤔